src/components/search-result.component.js

```diff
@@ -7,11 +7,115 @@ export const SearchResultComponent = {
   },
   controller: class SearchResult {
 
-    constructor($sce) {
+    constructor($sce,$http,$httpParamSerializerJQLike) {
       'ngInject';
+      this.IsVisible = false;
+      this.$http = $http;
+      this.$httpParamSerializerJQLike = $httpParamSerializerJQLike;
+      this.$state = {
+        // common contacts tab data
+        contacts: this.CONTACTS_LIST,
+
+        // directory API data
+        departments: null,
+        users: null,
+        usersCache: null,
+
+        // data and status of contact form
+        formData: {
+          recipient: null,
+          sendCopy: null,
+          bannerId: null,
+          firstname: null,
+          lastname: null,
+          department: null,
+          position: null,
+          building: null,
+          office: null,
+          extension: null,
+          email: null
+        },
+        formStatus: {
+          success: null,
+          error: null
+        },
+
+        // pagination
+        currentPage: 0,
+        pageSize: 10,
+
+        //sorting
+        sortOptions: this.SORT_OPTIONS,
+        sortOrder: 'lastname',
+
+        // whether results are currently loading
+        loadingResults: false,
+      };
+      /**
+       * A map of DOM element IDs as string constants.
+       */
+      this.ID = {
+        SEARCH_FORM: 'search-form',
+        SEARCH_RESULTS: 'search-results',
+        DIRECTORY_TABS: 'directory-tabs',
+        DIRECTORY_TABS_CONTENT: 'directory-tabs-content'
+      };
+
       if (this.result && this.result.expert && this.result.expert.summary) {
         this.result.expert.summary = $sce.trustAsHtml(this.result.expert.summary);
       }
     }
+
+
+    /**
+     * hide and show the contact form with the click of button.
+     */
+    showHide() {
+      this.IsVisible = this.IsVisible ? false : true;
```

Comment:
First letter lowercase for camelcasing, `.isVisible`

Reply:
oh well~